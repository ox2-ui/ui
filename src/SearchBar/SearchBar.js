import PropTypes from 'prop-types';
import React from 'react';
import SearchInput from 'react-search-input';
import styled from 'styled-components';

const StyledSearchInput = styled(SearchInput)`
  input {
    width: 100%;
    font-size: 15px;
    border: 1px solid #c6c6c6;
    border-radius: 3px;
    line-height: 22px;
    padding-left: 9px;
    padding-right: 9px;
    height: 32px;
    -webkit-appearance: none;
    font-family: Roboto Condensed
  }

  input:focus {
    outline: none;
  }
`;

/**
 * SearchBar Component
 */
const SearchBar = ({
  className,
  style,
  onChange,
  placeholder,
  value,
  ...other
}) => {
  return (
    <StyledSearchInput
      className={className}
      onChange={onChange}
      placeholder={placeholder}
      style={style}
      throttle={100}
      value={value}
      {...other}
    />
  );
};

SearchBar.propTypes = {
  className: PropTypes.string,
  onChange: PropTypes.func,
  placeholder: PropTypes.string,
  style: PropTypes.object,
  value: PropTypes.string,
};

export default SearchBar;
