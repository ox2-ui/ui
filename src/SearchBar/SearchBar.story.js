import React from 'react';
import { storiesOf } from '@kadira/storybook';
import SearchBar from './SearchBar';

const onSearchUpdate = (searchTerm) => console.log('🍇', searchTerm); // eslint-disable-line no-console

storiesOf('SearchBar', module)
  .add('default', () => (
    <SearchBar
      onChange={onSearchUpdate}
      placeholder={'Search this...'}
    />
  ))
  .add('controlled', () => (
    <SearchBar
      onChange={onSearchUpdate}
      placeholder={'Search this...'}
      value={'some'}
    />
  ))
  .add('autoFocus', () => (
    <SearchBar
      autoFocus={true}
      onChange={onSearchUpdate}
      placeholder={'Search this...'}
      value={'some'}
    />
  ));
