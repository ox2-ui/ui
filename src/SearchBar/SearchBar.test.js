/* eslint-disable no-unused-vars */
import React from 'react';
import { shallow, render, mount } from 'enzyme';
import SearchBar from './SearchBar';

const onSearchUpdate = () => console.log('🍇'); // eslint-disable-line no-console

it('renders correctly', () => {
  const wrapper = mount(
    <SearchBar
      onChange={onSearchUpdate}
      placeholder={'Search this...'}
    />,
  );

  expect(wrapper).toMatchSnapshot();
});
