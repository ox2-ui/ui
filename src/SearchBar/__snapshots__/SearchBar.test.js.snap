// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<SearchBar
  onChange={[Function]}
  placeholder="Search this..."
>
  <Styled(Search)
    onChange={[Function]}
    placeholder="Search this..."
    throttle={100}
  >
    <Search
      caseSensitive={false}
      className="dfnEEL"
      fuzzy={false}
      onChange={[Function]}
      placeholder="Search this..."
      throttle={100}
    >
      <div
        className="dfnEEL"
      >
        <input
          onChange={[Function]}
          placeholder="Search this..."
          type="search"
          value=""
        />
      </div>
    </Search>
  </Styled(Search)>
</SearchBar>
`;
